File: codefights/go/knapsack_light/code.go
func knapsackLight(value1 int, weight1 int, value2 int, weight2 int, maxW int) int {
    return_value := 0
	if (weight1 <= maxW) && (weight2 <= maxW) {
		if (weight1 + weight2) <= maxW {
			return_value += (value1 + value2)
		} else {
			switch {
			case (value2 > value1):
				return_value += value2
				break
			case (value1 > value2) || (value1 == value2):
				return_value += value1
				break
			}
		}
	} else if weight1 > maxW {
		switch {
		case (weight1 <= maxW):
			return_value += value2
			break
		case (weight2 == maxW):
			return_value += value2
			break
		}
	} else if weight2 > maxW {
		if weight1 <= maxW {
			return_value += value1
		}
	}
	return return_value
}

File: codefights/elixir/knapsack_light/code.ex
def knapsackLight(value1, weight1, value2, weight2, maxW) do
    cond do
        (weight1 <= maxW) && (weight2 <= maxW) ->
            cond do
                (weight1 + weight2) <= maxW -> value1 + value2
                (weight1 + weight2) > maxW ->
                        cond do
                            value2 > value1 -> value2
                            value1 > value2 -> value1
                            value1 == value2 -> value1
                        end
            end
        (weight1 > maxW) -> 
            cond do
                (weight2 <= maxW) -> value2
                (weight2 > maxW) -> 0
            end
        (weight2 > maxW) ->
            cond do
                (weight1 <= maxW) -> value1 
            end
    end
end
